lib/utils/normalizeOptions.js

```diff
@@ -91,6 +95,15 @@ function normalizeOptions(compiler, options, logger) {
       options: defaultWebSocketServerOptions,
     };
   } else {
+    if (
+      options.webSocketServer.options &&
+      typeof options.webSocketServer.options.port === 'string'
+    ) {
+      options.webSocketServer.options.port = Number(
+        options.webSocketServer.options.port
+      );
+    }
```

Comment:
Make sense to simplify it moving here https://github.com/webpack/webpack-dev-server/blob/5adf5d3a4de11451ff65c00d63c0d89d6f4e2ddf/lib/utils/normalizeOptions.js#L113 and use

```js
// i.e. normalize after normalize :)
if (typeof options.webSocketServer.port === 'string') {
  options.webSocketServer.port = Number(options.webSocketServer.port);
}
```